File: app/src/main/java/ru/eva/oriokslive/activities/splash/SplashActivity.java
package ru.eva.oriokslive.activities.splash;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import ru.eva.oriokslive.R;
import ru.eva.oriokslive.activities.main.MainActivity;
import ru.eva.oriokslive.activities.registration.RegistrationActivity;

public class SplashActivity extends AppCompatActivity implements ContractSplashActivity.View {

    private ContractSplashActivity.Presenter mPresenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        mPresenter = new PresenterSplashActivity(this);
        mPresenter.checkAccessToken();
    }

    @Override
    public void startMainActivity() {
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }

    @Override
    public void startRegistrationActivity() {
        startActivity(new Intent(this, RegistrationActivity.class));
        finish();
    }

    @Override
    public void showToast(String error) {
        Toast.makeText(this, error, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/ru/eva/oriokslive/ui/activity/splash/SplashActivity.kt
package ru.eva.oriokslive.ui.activity.splash

import android.content.Intent
import android.view.LayoutInflater
import android.widget.Toast
import androidx.activity.viewModels
import dagger.hilt.android.AndroidEntryPoint
import ru.eva.oriokslive.R
import ru.eva.oriokslive.databinding.ActivitySplashBinding
import ru.eva.oriokslive.ui.activity.main.MainActivity
import ru.eva.oriokslive.ui.activity.registration.RegistrationActivity
import ru.eva.oriokslive.ui.base.BaseActivity

@AndroidEntryPoint
class SplashActivity : BaseActivity<ActivitySplashBinding>() {

    override val bindingInflater: (LayoutInflater) -> ActivitySplashBinding =
        ActivitySplashBinding::inflate

    private val viewModel: SplashViewModel by viewModels()

    override fun setupUI() {
        viewModel.checkAccessToken()
        viewModel.startActivity.observe(this) {
            startActivity(Intent(this, MainActivity::class.java))
            finish()
        }
        viewModel.onError.observe(this) {
            Toast.makeText(this, R.string.token_deleted, Toast.LENGTH_SHORT).show()
            startActivity(Intent(this, RegistrationActivity::class.java))
            finish()
        }
    }
}